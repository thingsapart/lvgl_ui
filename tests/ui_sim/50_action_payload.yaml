# TICKS: 5
- type: data-binding
  state:
    - from_float: 0.0
    - from_bool: false
    - from_string: string
  actions:
    - set_f:
        set: { from_float: value.float }
    - set_b:
        set: { from_bool: value.bool }
    - set_s:
        set: { from_string: value.string }
  schedule:
    - { tick: 1, action: set_f, with: 123.45 }
    - { tick: 2, action: set_b, with: true }
    - { tick: 3, action: set_s, with: "testing" }
    - { tick: 4, action: set_b, with: false }
    - { tick: 5, action: set_s, with: "" }
