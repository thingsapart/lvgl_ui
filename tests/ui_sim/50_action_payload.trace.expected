--- UI-Sim Trace Start ---
STATE_SET: from_float = 0.000 (old: null)
STATE_SET: from_bool = false (old: null)
STATE_SET: from_string = "" (old: null)
NOTIFY: from_float = 0.000
NOTIFY: from_bool = false
NOTIFY: from_string = ""

--- TICK 1 ---
ACTION: set_f value=123.450
STATE_SET: from_float = 123.450 (old: 0.000)
NOTIFY: from_float = 123.450

--- TICK 2 ---
ACTION: set_b value=true
STATE_SET: from_bool = true (old: false)
NOTIFY: from_bool = true

--- TICK 3 ---
ACTION: set_s value="testing"
STATE_SET: from_string = "testing" (old: "")
NOTIFY: from_string = "testing"

--- TICK 4 ---
ACTION: set_b value=false
STATE_SET: from_bool = false (old: true)
NOTIFY: from_bool = false

--- TICK 5 ---
ACTION: set_s value=""
STATE_SET: from_string = "" (old: "testing")
NOTIFY: from_string = ""

--- UI-Sim Trace End ---
